src/errors.rs

```diff
@@ -25,15 +29,32 @@ impl From<ffi::NulError> for TraceMeError {
     }
 }
 
+impl From<io::Error> for TraceMeError {
+    fn from(err: io::Error) -> Self {
+        TraceMeError::IO(err)
+    }
+}
+
+impl From<num::ParseIntError> for TraceMeError {
+    fn from(err: num::ParseIntError) -> Self {
+        TraceMeError::NumParseInt(err)
+    }
+}
+
 impl Display for TraceMeError {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         match self {
             &TraceMeError::FFIIntoString(ref e) => write!(f, "{}", e),
             &TraceMeError::FFINul(ref e) => write!(f, "{}", e),
+            &TraceMeError::IO(ref e) => write!(f, "{}", e),
+            &TraceMeError::NumParseInt(ref e) => write!(f, "{}", e),
```

Comment:
For all of these cases, you might want to merge them using `|`?

Reply:
Will it work? The `e`s have different types.

Reply:
Ah, I hadn't though of that. They probably won't work... but would be interesting to try!

Reply:
They all implement `Error`. So maybe...

Reply:
If `Error` implements the appropriate formatter, it will probably work I think.

Reply:
No cigar.

```rust
        match self {    
            &TraceMeError::FFIIntoString(ref e)                           
            | &TraceMeError::FFINul(ref e)                           
            | &TraceMeError::IO(ref e)                                          
            | &TraceMeError::NumParseInt(ref e) => write!(f, "{}", e),    
```

```
error[E0308]: mismatched types
  --> src/errors.rs:48:37
   |
48 |             | &TraceMeError::FFINul(ref e)
   |                                     ^^^^^ expected struct `std::ffi::IntoStringError`, found struct `std::ffi::NulError`
   |
   = note: expected type `&std::ffi::IntoStringError`
              found type `&std::ffi::NulError`

error[E0308]: mismatched types
  --> src/errors.rs:49:33
   |
49 |             | &TraceMeError::IO(ref e)
   |                                 ^^^^^ expected struct `std::ffi::IntoStringError`, found struct `std::io::Error`
   |
   = note: expected type `&std::ffi::IntoStringError`
              found type `&std::io::Error`

error[E0308]: mismatched types
  --> src/errors.rs:50:42
   |
50 |             | &TraceMeError::NumParseInt(ref e) => write!(f, "{}", e),
   |                                          ^^^^^ expected struct `std::ffi::IntoStringError`, found struct `std::num::ParseIntError`
   |
   = note: expected type `&std::ffi::IntoStringError`
              found type `&std::num::ParseIntError`

error: aborting due to 3 previous errors
```

Also tried annotating `e`. You can't do that it seems.

Reply:
OK, sorry for the noise!